src/realm/parser/driver.cpp

```diff
@@ -797,12 +797,49 @@ Query GeoWithinNode::visit(ParserDriver* drv)
     else {
```

Comment:
Why else here? There is a return above

Reply:
True. I restructured this to simplify the structure.